app/views/articles/_comment_tree.html.erb

```diff
@@ -1,2 +1,4 @@
 <% comment, sub_comments = comment_node %>
-<%= nested_comments(tree: { comment => sub_comments }, commentable: @article, is_view_root: true) %>
+<% unless comment.decorate.low_quality && sub_comments.empty? %>
```

Comment:
Added logic here and not in `article_comment_tree` because it should reduce the number of queries while checking for children. However, I'm not a fan of this solution, open for suggestions.